frontend/packages/topology/src/actions/provider.ts

```diff
@@ -17,3 +22,39 @@ export const useTopologyWorloadActionProvider = (element: GraphElement) => {
     return [actions, true, undefined];
   }, [actions, element]);
 };
+
+export const useTopologyContextActionProvider = (element: GraphElement) => {
+  const [namespace] = useActiveNamespace();
+  const [
+    addActionExtensions,
+    addActionExtensionsResolved,
+    allAddActionsDisabled,
+  ] = useAddActionExtensions();
+  const [
+    filteredAddActionExtensions,
+    filteredAddActionExtensionsLoaded,
+  ] = useAccessFilterExtensions(
+    namespace,
+    filterNamespaceScopedUrl(namespace, addActionExtensions),
+  );
+  const loaded =
+    !allAddActionsDisabled && addActionExtensionsResolved && filteredAddActionExtensionsLoaded;
+
+  const loadedActions = useMemo(
+    () => loaded && filteredAddActionExtensions.filter((a) => !!a.properties.action),
+    [loaded, filteredAddActionExtensions],
+  );
+
+  return useMemo(() => {
+    if (element.getKind() === ModelKind.graph) {
+      const actions = loaded && loadedActions.map((a) => contextAction(a, namespace));
+      return [actions, loaded, undefined];
+    }
+    if (element.getType() === TYPE_APPLICATION_GROUP) {
```

Comment:
you can add deleteAction in an array with other application actions if you create a separate action provider